internal/http/graphql_client.go

```diff
@@ -0,0 +1,79 @@
+package http
+
+import (
+	"strings"
+
+	"github.com/newrelic/newrelic-client-go/internal/logging"
+	"github.com/newrelic/newrelic-client-go/internal/region"
+	"github.com/newrelic/newrelic-client-go/pkg/config"
+)
+
+// GraphQLClient represents a graphQL HTTP client.
+type GraphQLClient struct {
+	client *NewRelicClient
+	config config.Config
+	logger logging.Logger
+}
+
+// NewGraphQLClient returns a new instance of GraphQLClient.
+func NewGraphQLClient(cfg config.Config) *GraphQLClient {
+	if cfg.BaseURL == "" {
+		cfg.BaseURL = region.NerdGraphBaseURLs[region.Parse(cfg.Region)]
+	}
+
+	c := NewClient(cfg)
+	c.SetErrorValue(&graphQLErrorResponse{})
+
+	return &GraphQLClient{
+		client: &c,
+		config: cfg,
+		logger: cfg.GetLogger(),
+	}
+}
+
+// Query runs a graphQL query.
+func (g *GraphQLClient) Query(query string, vars map[string]interface{}, respBody interface{}) error {
+	req := graphQLRequest{
+		Query:     query,
+		Variables: vars,
+	}
+
+	resp := graphQLResponse{
+		Data: respBody,
+	}
+
+	if _, err := g.client.Post(g.config.BaseURL, nil, &req, &resp); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type graphQLRequest struct {
+	Query     string                 `json:"query"`
+	Variables map[string]interface{} `json:"variables,omitempty"`
+}
+
+type graphQLResponse struct {
+	Data interface{} `json:"data"`
+}
+
+type graphQLError struct {
+	Message string `json:"message"`
```

Comment:
Thoughts on including the query `path` from the error? `Path []string `json:"path"``

i.e.
```
"message": "Entity guid required",
{
  //...
      "message": "Entity guid required"
      "path": [
        "actor",
        "entity"
      ]
}
```

Reply:
Would you want to include it in the error message returned to the user?  Right now we are just doing a `strings.Join` on all the messages inside the returned errors, and returning that. 

https://github.com/newrelic/newrelic-client-go/pull/95/files#diff-f821ec257b1e058ba8dc4ba07b6edeceR75

Reply:
Seems like that could be a noisy error response and potentially a little harder to read, but I haven't seen what the actual output looks like so hard to say. I think it would be a separate field in the current struct you have, but if it's a pain or involves anything more than a conditional check to see if `path` is present in the error object (not sure if its always present I guess), then it's not a big deal.

```go
type graphQLError struct {
	Message string `json:"message"`
  Path []string `json:"path"`
}
```

 I could see this as potentially helpful with debugging.

Reply:
If it's unreasonable overhead to add to this struct
```go
type graphQLErrorResponse struct {
	Errors []graphQLError `json:"errors"`
}
```
then no worries.